### ISSUE TITLE
boostraping hvac-cli

### ISSUE THREAD

**Issue by singuliere** [hvac/hvac-cli#1]
@jeffwecan thanks for creating this repository. To get started would you mind if I push directly to master? As soon as it's more or less working and no later than in a week from now, I'll switch to submitting pull requests. Unless I'm mistaken I don't have permission to push right now. If you would rather start with pull request from the beginning it's fine by me though, just let me know.

- [x] setting up https://hvac-cli.readthedocs.io with a webhook from this repo
- [x] setting up https://travis-ci.com with a webhook from this repo
- [x] setting up PyPI project for hvac-cli

**Comment by singuliere** [comment]
> Unless I'm mistaken I don't have permission to push right now.

I did not notice the invitation received by mail, sorry for the noise :-)

**Comment by singuliere** [comment]
@jeffwecan the master branch has the basic structure of the hvac-cli command line, implementing kv get and put. It also has auto generated RTD doc using the cliff sphinx plugin. The tests use a vault fixture that runs a dev server with TLS.

I'm inclined to stay as close to the `vault` cli as possible, where relevant (i.e. not mimicing the unusual single dash long option ;-).

What do you think?

**Comment by singuliere** [comment]
I'm done with the bootstraping phase and ready to switch to the regular pull request / review mode :tada:

**Comment by singuliere** [comment]
@jeffwecan I'm happy to create the pypi & readthedocs projects, if you think it's ok?

**Comment by jeffwecan** [comment]
Sure thing! Just add me as a collaborator on those services if you don't mind. :D

I haven't had too much time as of late for side projects but I'm excited to check out what you've added thus far when I get the chance.

**Comment by singuliere** [comment]
> Sure thing! Just add me as a collaborator on those services if you don't mind. :D

Absolutely!

> I haven't had too much time as of late for side projects but I'm excited to check out what you've added thus far when I get the chance.

I'll keep pushing to master then, instead of spamming you with pull requests ;-) Let me know as soon as you're willing to get involved and I'll switch to pull request mode.

**Comment by singuliere** [comment]
@jeffwecan  I'm blocked with travis because of insufficient permissions. The [RTD is ready though](https://hvac-cli.readthedocs.io/) :tada:

![a](https://user-images.githubusercontent.com/35190819/53422186-47680380-39df-11e9-90f8-23b5d1097714.png)

**Comment by singuliere** [comment]
@jeffwecan the user **jeffwecan** does not exist on RTD, what is your account there so I can add you as a co-maintainer?

**Comment by jeffwecan** [comment]
Ah yes, for RTD I'm under *futuernorn*.

I've attempted to configure the Travis CI integration for this repository but I'm not certain if it is all set or not: https://travis-ci.com/hvac/hvac-cli

For what it is worth, thus far we've still been using travis-ci.org under the hvac repo since that is what it was set up with initially and Travis didn't have a migration path from .org to .com last time I checked. So I'm unfamiliar with the specifics on the usage of the travis-ci.com Github integration.

In any case, if you're unable to get builds running under https://travis-ci.com/hvac/hvac-cli once we have a .travis.yml config file in this codebase, I'd be fine expanding your access on the organization level so you can triage it directly. 👍

**Comment by singuliere** [comment]
> Ah yes, for RTD I'm under futuernorn.

You're a co-maintainer

> unfamiliar with the specifics on the usage of the travis-ci.com Github integration.

I'm not either, I'll add a .travis.yml and see what happens.

**Comment by singuliere** [comment]
@jeffwecan on second thought, I did not mean to use https://travis-ci.com/ but https://travis-ci.org/. I was not aware they were two different things. So... if you don't mind switching back to travis-ci.org, it will be in line with what you're currently using and presumably more friendly to our volunteer effort on hvac ;-)

**Comment by jeffwecan** [comment]
Well shucks, from what I can see it looks like the ability to add new repositories for travis-ci.org may have been shuttered:

> Note: GitHub Services have been deprecated. Please contact your integrator for more information on how to migrate or replace a service with webhooks or GitHub Apps.

I.e., the extant travis-ci.org / https://github.com/hvac/hvac integration is accomplished via GitHub Services config. So presumably its _hvac_ that will need to get migrated over to travis-ci.com at some point in the future...

**Comment by singuliere** [comment]
> Well shucks, from what I can see it looks like the ability to add new repositories for travis-ci.org may have been shuttered:

Shoot. Oh well, I pushed `.travis.yml`, let see how travis-ci.com reacts.

**Comment by singuliere** [comment]
And travis is happy, badge added: we're green with decent coverage (98%).

**Comment by singuliere** [comment]
@jeffwecan what's your PyPI account name?

**Comment by jeffwecan** [comment]
Should be plain 'ole *jeffwecan* on the PyPi front. Related, I've found it handy to also utilize a https://test.pypi.org/ repository. Especially useful for a package's initial publishing or when making any type of description changes (since you can't really unpublish releases to PyPi proper).

**Comment by singuliere** [comment]
Yeah, I recently discovered this test repo and https://devpi.net/. Since nobody is a aware of hvac-cli right now I'll push to PyPI directly, unless you think it's a bad idea?

**Comment by jeffwecan** [comment]
Nah, that sounds fine for now. I'm speaking more from my experience of inadvertently breaking folks usage of an existing module and not having a good way to rectify that fact. :D

**Comment by jeffwecan** [comment]
Oh and now that I've read through this issue's content a bit more, I have couple bits of feedback:

* We potentially have cause to work back on this (relatively) ancient hvac issue requesting the Vault integration test harness bits be made reusable via something like a nosetests plugin: https://github.com/hvac/hvac/issues/99. At the very least, if there ends up being enough similar functionality between this repo and hvac, it would make sense to look into. (Though I've also chatted with a friend who have done similar API client + cli tool backed by that client that have good experiences keeping the two pieces of code within the same repo. 👋 @dbwpe) 
* On the sphinx front, I've actually spent the bulk of my hvac development time over the past month or so implementing doctest after receiving a good deal of documentation typo reports. So I'd recommend checking out that extension if you're interested. I have been pretty pleased with it thus far! https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html
